java/arcs/core/host/AbstractArcHost.kt

```diff
@@ -535,36 +460,16 @@ abstract class AbstractArcHost(
      * releasing [Handle]s, and modifying [ArcState] and [ParticleState] to stopped states.
      */
     private suspend fun stopArcInternal(arcId: String, context: ArcHostContext) {
-        val scheduler = schedulerProvider(arcId)
-        withContext(scheduler.asCoroutineDispatcher()) {
-            context.particles.values.forEach { particleContext -> stopParticle(particleContext) }
+        context.particles.values.forEach { particleContext ->
+            particleContext.stopParticle()
+            particleContext.particle.handles.reset()
         }
-        scheduler.waitForIdle()
         maybeCancelResurrection(context)
-        setArcState(context, ArcState.Stopped)
+        context.arcState = ArcState.Stopped
         updateArcHostContext(arcId, context)
         context.entityHandleManager.close()
```

Comment:
I feel like it might be worth putting the entityHandleManager.close() in a finally {} block, since updateArcHostContext in the past has thrown exceptions in serialization. This'll lead to a memory leak if it happens.